### DISCUSSION TITLE
Supabase Authentication: Why Doesn't SignUp with Existing Account Log In When 'autoconfirm' Is ON?

### DISCUSSION THREAD

**Discussion by vladimirzb** [supabase/supabase#16760] [Questions] [answered]
I'm using the `supabase.auth.signUp` function in my code, expecting a logged-in session when the server has "autoconfirm" ON. However, when signing up with an existing account, it doesn't sign in as expected. Here's the code snippet:

```javascript
const handleSignUp = async () => {
  await supabase.auth.signUp({
    email,
    password,
    options: {
      emailRedirectTo: `${location.origin}/auth/callback`,
    },
  });
  router.refresh();
}
```

Is the default configuration of Supabase's "autoconfirm" set to off? If so, where can I find and change this configuration? If the default is on, what could be the issue causing this behavior? Any insights would be greatly appreciated.

**Comment by GaryAustin1** [comment, marked as answer]
If you signUp and have an existing user email, then Supabase returns fake info to try an avoid a security leak.

The confirm setting only applies to a new user.
![image](https://github.com/supabase/supabase/assets/54564956/eea228b5-2855-479a-930c-0a3afd4eb467)

If on then you will get redirected if user is new and they need to confirm.
If off then you get a real user session back, or a fake user if they already exist.

SignUp will never log in an existing user.

**Reply by vladimirzb** [reply]
Thank you for the explaination!
